### ISSUE TITLE
Issue found by Svace static analyzer

### ISSUE THREAD

**Issue by vvdvortsova** [etcd-io/etcd#13176]
Hello
This warning found with the Svace.
In each case, using a variable remoteServer after comparison with nil, remoteServer may be nil. It may lead to panic.
https://github.com/etcd-io/etcd/blob/af9b5e726a1c079156725432b22b54371ee12a69/server/etcdserver/api/rafthttp/http.go#L464
[dereference] Variable 'remoteServer' is passed to function 'rafthttp.checkVersionCompatibility' as 2nd parameter at http.go:474
https://github.com/etcd-io/etcd/blob/af9b5e726a1c079156725432b22b54371ee12a69/server/etcdserver/api/rafthttp/http.go#L474
Variable 'server' is passed to function 'rafthttp.compareMajorMinorVersion' as 1st parameter at util.go:175
https://github.com/etcd-io/etcd/blob/af9b5e726a1c079156725432b22b54371ee12a69/server/etcdserver/api/rafthttp/util.go#L175
a is dereferenced here. at util.go:135
https://github.com/etcd-io/etcd/blob/af9b5e726a1c079156725432b22b54371ee12a69/server/etcdserver/api/rafthttp/util.go#L135

**Comment by nic-chen** [comment]
hi @vvdvortsova 

It seems that `remoteServer` cannot be nil, otherwise it will panic in `semver.Must`:

https://github.com/etcd-io/etcd/blob/af9b5e726a1c079156725432b22b54371ee12a69/server/etcdserver/api/rafthttp/util.go#L154

**Comment by vvdvortsova** [comment]
> hi @vvdvortsova
> 
> It seems that `remoteServer` cannot be nil, otherwise it will panic in `semver.Must`:
> 
> https://github.com/etcd-io/etcd/blob/af9b5e726a1c079156725432b22b54371ee12a69/server/etcdserver/api/rafthttp/util.go#L154
Yes, I agree with you. Therefore it's a redundant comparison with nil. Can I send a PR to fix it?
